src/server/status_server/mod.rs

```diff
@@ -495,6 +502,27 @@ where
         }
     }
 
+    fn handle_get_metrics(req: Request<Body>) -> hyper::Result<Response<Body>> {
+        let gz_encoding = client_accept_gzip(&req);
+        let metrics = if gz_encoding {
+            // gzip can reduce the body size to less than 1/10.
+            let mut encoder = GzEncoder::new(vec![], Compression::default());
+            dump_to(&mut encoder);
+            encoder.finish().unwrap()
+        } else {
+            dump().into_bytes()
+        };
+        let mut resp = Response::new(metrics.into());
+        resp.headers_mut()
```

Comment:
Why not combine this branch with the `if` below?

Reply:
the `CONTENT-TYPE` header is always set, `CONTENT-ENCODING` is set only when it is gzip.